src/nats/mod.rs

```diff
@@ -0,0 +1,222 @@
+use std::borrow::Cow;
+
+use testcontainers::{core::WaitFor, Image, RunnableImage};
+
+/// Nats image for [testcontainers](https://crates.io/crates/testcontainers).
+///
+/// This image is based on the official [Nats](https://hub.docker.com/_/nats) image.
+/// The default user is `` and the default password is ``.
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Nats {
+    version: Value,
+    user: Option<Value>,
+    pass: Option<Value>,
+}
+
+impl Nats {
+    const DEFAULT_USER: &'static str = "";
+    const DEFAULT_PASS: &'static str = "";
+    const DEFAULT_VERSION_TAG: &'static str = "2.10.14";
+
+    /// Create a new instance of a Nats image.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            version: Cow::Borrowed(Self::DEFAULT_VERSION_TAG),
+            user: Some(Cow::Borrowed(Self::DEFAULT_USER)),
+            pass: Some(Cow::Borrowed(Self::DEFAULT_PASS)),
+        }
+    }
+
+    /// Set the username to use.
+    #[must_use]
+    pub fn with_user(mut self, user: impl Into<Value>) -> Self {
+        self.user = Some(user.into());
+        self
+    }
+
+    /// Set the password to use.
+    #[must_use]
+    pub fn with_password(mut self, pass: impl Into<Value>) -> Self {
+        self.pass = Some(pass.into());
+        self
+    }
+
+    /// Do not use any authentication on the testcontainer.
+    ///
+    /// Setting this will override any prior usages of [`Self::with_user`] and
+    /// [`Self::with_password`].
+    pub fn without_authentication(mut self) -> Self {
+        self.user = None;
+        self.pass = None;
+        self
+    }
+}
+
+type Value = Cow<'static, str>;
+
+impl Default for Nats {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// The actual Nats testcontainers image type which is returned by `container.image()`
+pub struct NatsImage {
+    version: String,
+    auth: Option<(String, String)>,
+}
```

Comment:
I think it's unnecessary complication to have a builder with `Cow` value wrappers. 

I'd prefer to provide a single struct `Nats`, which has builder-methods (accepts `impl Into<String>`) and implements `Image`.

Couple of reasons:
1. Keep as simple as possible
1.1 Only single public structure exposed, it provides everything we need
2. There is no real benefits, since we convert `Cow` to owned strings anyway. 
2.1. Empty string doesn't require allocation
2.2. Also, performance is not that important for testing, docker overhead is much more significant.
3. There is no need for explicit `impl From<Nats> for RunnableImage<NatsImage>`

So I suggest to have:

```rs
pub struct Nats {
    version: String,
    user: Option<String>,
    pass: Option<String>,
}

impl Nats {
 // all builder-methods and getters 
}

impl Default for Nats {..}

impl Image for Nats {..}

```

Reply:
you are correct. I took the neo4j integration as a template to follow. Simplification made done : [bca942f](https://github.com/testcontainers/testcontainers-rs-modules-community/pull/123/commits/bca942f0af1f4f393a818ae132cc83c4b825415e)

Reply:
Thank you!

Yeah, good point, need to review Neo4J in order to simplify. Thanks for pointing that out